### PR TITLE
deps: use a fixed node.js version to prevent build issues with unknown versions

### PR DESCRIPTION
**Pull request by shauke**
* fixed to node.js 16.16.0 including npm 8.11.0

## PR Type

[x] Build-related changes

## What Is the Current Behavior?

docker and git actions use latest node.js 16 version that led to unexpected behavior

## What Is the New Behavior?

fixed node.js version does not introduce unexpected behavior/issues

## Does this PR Introduce a Breaking Change?

[x] No

## Other Information

[AB#78109](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/78109)

### CODE REVIEW

**Comment by jometzner** [comment]
Maybe we could introduce some variable for the used nodejs version in GitHub pipelines? Changing the version number everywhere is not a pleasant task.

**Comment by shauke** [comment]
I could either use a GitHub secret for the node version number, which is the easiest way but not very transparent since the actual value is hidden in the management view.

Or I could try to use this action to set a global variable that could be used by all workflows (https://github.com/marketplace/actions/set-environment-variables) but this is kind of much overhead for the problem at hand, but maybe just the one value to start with for more to come?

Or we could leave it to be changed everywhere. ;)  

What is the preferred action?

**Comment by jometzner** [comment]
> I could either use a GitHub secret for the node version number, which is the easiest way but not very transparent since the actual value is hidden in the management view.
> 
> Or I could try to use this action to set a global variable that could be used by all workflows (https://github.com/marketplace/actions/set-environment-variables) but this is kind of much overhead for the problem at hand, but maybe just the one value to start with for more to come?
> 
> Or we could leave it to be changed everywhere. ;)
> 
> What is the preferred action?

Yes, I noticed that github has no such thing as variable groups that can be referred inside the workflow. I guess the best compromise is to use a environment variable. Scope it using the entire workflow like so:
```
on:
  bla

env:
  NODEJS_VERSION: 16.16.0

jobs:
```
I'll commit an example.

File: docs/guides/migrations.md [hunk 1]
```diff
@@ -9,6 +9,11 @@ kb_sync_latest_only
 
 ## 2.4 to 3.0
 
+With the 2.4.1 Hotfix we introduced a more fixed Node.js version handling.
+For the docker containers and as the projects engine we set `Node.js 16.16.0` and `npm 8.11.0` to more specifically refer to the versions used and tested.
```
Comment (jometzner) [inline docs/guides/migrations.md:13]: Sorry I have trouble understanding that sentence. May I suggest:
`We set Node.js 16.16.0 and npm 8.11.0 as our application runtime and package management versions.`

**Pull request merged**